src/fromager/sources.py

```diff
@@ -326,17 +335,21 @@ def _apply_patch(patch: pathlib.Path, source_root_dir: pathlib.Path):
 
 
 def _warn_for_old_patch(
-    source_root_dir: pathlib.Path, patch_filename: pathlib.Path
+    source_root_dir: pathlib.Path,
+    patches_dir: pathlib.Path,
 ) -> None:
-    # Get the existing version and req name as per the source_root_dir naming conventions
+    # Get the req name as per the source_root_dir naming conventions
     req_name = source_root_dir.name.rsplit("-", 1)[0]
-    existing_version = Version(source_root_dir.name.rsplit("-", 1)[1])
 
-    # Check to see if patches for older version exists
-    if patch_filename.parent.name == source_root_dir.name:
-        logger.warning(
-            f"{req_name}: patches for version {existing_version!s} exist but will not be applied"
-        )
+    # Filter the patches using regex
+    patch_directories = overrides.get_patch_directories(patches_dir, source_root_dir)
+    current_path = source_root_dir.parent.joinpath("patches_dir", source_root_dir.name)
+    # If filtered patches contain patches for current version, remove the path because we dont want to warn
+    if current_path in patch_directories:
+        patch_directories.remove(current_path)
```

Comment:
I don't think filtered patches will contain the current version since you are only calling this function if `has_applied` is false which implies that were no patches directory for the current version.

You can add a docstring for this helper to clarify this point so that any other caller can use it correctly

Reply:
Okay, I added it as docstring